inc/search.class.php

```diff
@@ -1227,21 +1314,9 @@ static function constructData(array &$data, $onlycount = false) {
          }
 
          $data['data']['totalcount'] = 0;
-         // if real search or complete export : get numrows from request
-         if (!$data['search']['no_search']
-             || $data['search']['export_all']) {
-            $data['data']['totalcount'] = $DBread->numrows($result);
-         } else {
-            if (!isset($data['sql']['count'])
-               || (count($data['sql']['count']) == 0)) {
-               $data['data']['totalcount'] = $DBread->numrows($result);
-            } else {
-               foreach ($data['sql']['count'] as $sqlcount) {
-                  $result_num = $DBread->query($sqlcount);
-                  $data['data']['totalcount'] += $DBread->result($result_num, 0, 0);
-               }
-            }
-         }
+         $result_num = $DBread->query($data['sql']['count']);
+         $data['data']['totalcount'] += $DBread->result($result_num, 0, 0);
+         $data['data']['count'] = intval(min($data['data']['totalcount'], $data['search']['list_limit']));
```

Comment:
Why there is a `intval` here ? If `$data['data']['totalcount']` or `$data['search']['list_limit']` may be something else than an integer, it should be clearly explained in a comment. Also, I guess it would be prefferable to apply the `intval` transformation prior to the `min` call.

Reply:
Some tests were failing due to a string value in `$data['data']['count']`.
Min may return a string if the compared value are numbers stored in string: 
![image](https://user-images.githubusercontent.com/42734840/133767510-a5d2fa39-e083-47e3-b74d-d767ae573d6c.png)

I've moved intval to only target `$data['search']['list_limit']` since it is the offender.
`$data['data']['totalcount']` is defined just above so it should be safe to assume it'll always be a integer.